### PR TITLE
Feature : new implementation of DFT+U

### PR DESCRIPTION
**Pull request by wenfei-li**
For more details, check the comment in LCAO_dftu_new.h

### CODE REVIEW

**Comment by wenfei-li** [comment]
This functionality is unfinished. There is a problem related to passing kv.isk all the way down into folding_fixedH. It can be solved, but is rather cumbersome. In the meantime I'm also waiting for the completion of a test workflow. As a result, I'll leave it as is for now and resume on this project later.

**Comment by wenfei-li** [comment]
I've done SCF for multi-k (so no gamma point, no force & stress); the results seem to be reasonable. I will wait for the completion of test workflow and continue on this project if the new implementation is tested to be useful.

**Comment by wenfei-li** [comment]
I turned off coverage for this module because this functionality is not intended to be used as an official functionality but rather something to make experiments on. Another option would be to not merge this pr and use my dftu branch for the experiments. I'll wait for further discussion on this problem.

**Comment by hongriTianqi** [comment]
@wenfei-li @baixiaokuang Sorry that I cannot agree to merge this PR directly into the develop branch by turning off coverage and without unit tests. The reason is simple, without unit tests, one cannot make sure that the new feature won't be breaked by future PRs, and cannot make sure it won't break existing functions of the code as well.

File: source/module_hamilt_lcao/module_dftu_new/LCAO_dftu_new.cpp [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+#include "module_hamilt_pw/hamilt_pwdft/global.h"
+#include "module_base/parallel_common.h"
+
+namespace GlobalC
+{
+    LCAO_DftU_New lcao_dftu_new;
+}
+
+//Constructor of the class
+LCAO_DftU_New::LCAO_DftU_New()
```
Comment (hongriTianqi) [inline source/module_hamilt_lcao/module_dftu_new/LCAO_dftu_new.cpp:25]: Please add unit tests for the new module. Thanks.

**Review by hongriTianqi — CHANGES_REQUESTED** [review]

**Comment by wenfei-li** [comment]
alright then we'll use my dftu branch for testing purpose

**Comment by dyzheng** [comment]
I will test the dftu branch, set this PR to draft.

**Pull request closed without merging**